astropy/utils/masked/core.py

```diff
@@ -0,0 +1,1086 @@
+        # and do float16 at higher precision.
+        is_float16_result = False
+        if dtype is None:
+            if issubclass(self.dtype.type, (np.integer, np.bool_)):
+                dtype = np.dtype('f8')
+            elif issubclass(self.dtype.type, np.float16):
+                dtype = np.dtype('f4')
+                is_float16_result = out is None
+
+        result = self.sum(axis=axis, dtype=dtype, out=out,
+                          keepdims=keepdims, where=~self.mask)
+        n = np.add.reduce(~self.mask, axis=axis, keepdims=keepdims)
+        result /= n
+        if is_float16_result:
+            result = result.astype(self.dtype)
+        return result
+
+    def var(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+        # Simplified implementation based on that in numpy/core/_methods.py
+        n = np.add.reduce(~self.mask, axis=axis, keepdims=keepdims)[...]
+
+        # Cast bool, unsigned int, and int to float64 by default.
+        if dtype is None and issubclass(self.dtype.type,
+                                        (np.integer, np.bool_)):
+            dtype = np.dtype('f8')
+        mean = self.mean(axis=axis, dtype=dtype, keepdims=True)
+
+        x = self - mean
+        x *= x.conjugate()  # Conjugate just returns x if not complex.
+
+        result = x.sum(axis=axis, dtype=dtype, out=out,
+                       keepdims=keepdims, where=~x.mask)
+        n -= ddof
+        n = np.maximum(n, 0, out=n)
+        result /= n
+        result._mask |= (n == 0)
+        return result
+
+    def std(self, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
+        result = self.var(axis=axis, dtype=dtype, out=out, ddof=ddof,
+                          keepdims=keepdims)
+        return np.sqrt(result, out=result)
+
+    def __bool__(self):
+        # First get result from array itself; this will error if not a scalar.
+        result = super().__bool__()
+        return result and not self.mask
+
+    def any(self, axis=None, out=None, keepdims=False):
+        return np.logical_or.reduce(self, axis=axis, out=out,
+                                    keepdims=keepdims, where=~self.mask)
+
+    def all(self, axis=None, out=None, keepdims=False):
+        return np.logical_and.reduce(self, axis=axis, out=out,
+                                     keepdims=keepdims, where=~self.mask)
+
+    # Following overrides needed since somehow the ndarray implementation
+    # does not actually call these.
+    def __str__(self):
+        return np.array_str(self)
+
+    def __repr__(self):
+        return np.array_repr(self)
+
+    def __format__(self, format_spec):
+        string = super().__format__(format_spec)
+        if self.shape == () and self.mask:
+            n = min(3, max(1, len(string)))
+            return ' ' * (len(string)-n) + '\u2014' * n
+        else:
+            return string
+
+
+class Maskedrecord(np.recarray, MaskedNDArray, data_cls=np.recarray):
```

Comment:
Should this be named `MaskedRecarray` or `MaskedRecordArray`?

Reply:
Hmm, yes, why was this `record`? I really do not know what I was thinking! The scalar `np.record`? Anyway, I now changed it to `MaskedRecarray`.